onnxruntime/test/providers/cpu/tensor/scatter_nd_op_test.cc

```diff
@@ -210,7 +210,7 @@ TEST(ScatterNDOpTest, ScatterND_18_mul_long_shape) {
   // The linter complains if the line is split into multiple lines.
   test1.AddInput<float>("updates", {3, 2, 3, 1, 1, 1, 1}, {2.0f, 4.0f, 8.0f, 16.0f, 32.0f, 64.0f, 128.0f, 256.0f, 512.0f, 1024.0f, 2048.0f, 4096.0f, 8192.0f, 16384.0f, 32768.0f, 65536.0f, 131072.0f, 262144.0f});
   test1.AddOutput<float>("output", {2, 2, 3, 1, 1, 1, 1}, {1638.4f, 6553.6f, 26214.4f, 104857.6f, 419430.4f, 1677721.625f, 12.8f, 25.6f, 51.2f, 102.4f, 204.8f, 409.6f});
-  test1.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider, kOpenVINOExecutionProvider});
+  test1.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider, kOpenVINOExecutionProvider, kWebGpuExecutionProvider});
```

Comment:
Can you add a test for the multiple updates to a single index? You can add the SAM_float32 test case from jsep test suite.

Reply:
This test case results in shape such as array<vect4<u32>> which is not handled in the current code. The fix could be from setting the shape/stride.

Reply:
Since webgpu ep has a pin to cpu feature, should we remove the duplicate indices check and instead tell customers to pin scatternd to cpu when they run into the duplicate indices edge case? Then we can investigate why their model exports scatternd in undefined behavior case?

@fs-eire @guschmue thoughts?

Reply:
We are not covering the duplicate indices in this PR.

Reply:
If we are not covering duplicate indices, then we need to remove the hasDuplicates check as it will cause performance bottleneck for models with reduction=none. I'd suggest basing the webgpu kernel off the jsep scatternd kernel before I adjusted it for this edge case.

Reply:
for the duplicate indices - I think the spec says that those are not allowed but we know about 1 model that does this and we know cpu can handle it.
Is it just 1 model?

Reply:
SAM and Dexter's test model have hit this edge case, others may too